homeassistant/components/sensor/dht.py

```diff
@@ -39,9 +39,8 @@ def setup_platform(hass, config, add_devices, discovery_info=None):
         "DHT22": Adafruit_DHT.DHT22,
         "AM2302": Adafruit_DHT.AM2302
     }
-    sensor = available_sensors[config['sensor']]
-
-    pin = config['pin']
+    sensor = available_sensors.get(config.get('sensor'))
+    pin = config.get('pin')
```

Comment:
do you mean PIN = 'pin' ? it seems redundant. what about sensor, temperature, humidity ?